common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java

```diff
@@ -87,13 +94,34 @@ private void buildSchemaRegistryConfigs(Map<String, ?> configs) {
         validateAndSetJacksonDeserializationFeatures(configs);
         validateAndSetTags(configs);
         validateAndSetMetadata(configs);
+        validateAndSetUserAgent(configs);
+        validateAndSetSecondaryDeserializer(configs);
+    }
+
+    private void validateAndSetSecondaryDeserializer(Map<String, ?> configs) {
+        if (isPresent(configs, AWSSchemaRegistryConstants.SECONDARY_DESERIALIZER)) {
+            Object secondaryDeserializer = configs.get(AWSSchemaRegistryConstants.SECONDARY_DESERIALIZER);
+            if (secondaryDeserializer instanceof String) {
+                this.secondaryDeserializer = (String) secondaryDeserializer;
+            } else if (secondaryDeserializer instanceof Class) {
+                this.secondaryDeserializer = ((Class) secondaryDeserializer).getName();
+            } else {
+                throw new AWSSchemaRegistryException("Invalid secondary de-serializer configuration");
+            }
+        }
     }
 
     private void buildCacheConfigs(Map<String, ?> configs) {
         validateAndSetCacheSize(configs);
         validateAndSetCacheTTL(configs);
     }
 
+    private void validateAndSetUserAgent(Map<String, ?> configs) {
+        if (isPresent(configs, AWSSchemaRegistryConstants.USER_AGENT_APP)) {
```

Comment:
What if customer overrides this configuration, how do we prevent customer from passing junk value or a custom user agent ?

Reply:
We override this value in the serializer and de-serializer to replace any values explicitly set by configuration. Nevertheless, clients can still override and send garbage values if they really want to since we are just using the AWS SDK feature.

Reply:
wouldn't that mess up our data. Can we not just protect it or allow only defined ENUMs. We can handle Kinesis case also by allowing KCL/KPL strings but checking if they are allowed ENUMs.

Reply:
This library can also be used by different services like Lambda, Firehose etc. to serialize / deserialize. This config shouldn't become a bottleneck for adding a new enum. Instead the metrics can be derived from free-form text anyways.